src/code_8028DF00.c

```diff
@@ -234,16 +234,16 @@ void func_8028E3A0(void) {
     if (D_80150120) {
 
         if (gCupCourseSelection == CUP_COURSE_FOUR) {
-            D_80150124 = 5;
+            gMenuSelectionFromQuit = OPTIONS_MENU;
```

Comment:
I was confused by the OPTIONS_MENU... it's actually brining the player to the winning ceremony cutscene.

Reply:
D_800DC524 is getting set to gMenuSelectionFromQuit, then in thread5_game_logic's main loop, D_800DC50C is set to D_800DC524 if they aren't equal. func_80002684 has a switch statement on the value of D_800DC524 and case 5 (OPTIONS_MENU) calls load_ending_sequence_royalraceway().